filter_plugins/cisco_nxos.py

```diff
@@ -0,0 +1,370 @@
+import re
+
+def parseRanges(config):
+    config = config.splitlines()
+    vlan_list = []
+
+    for line in config:
+        if line.startswith("vlan ") and ("-" in line or "," in line):
+            # Remove "vlan " prefix
+            vlan_range_str = line.removeprefix("vlan ")
+
+            # This is a vlan with a range
+            for vlan_part in vlan_range_str.split(","):
+                if "-" in vlan_part:
+                    vlan_range = vlan_part.split("-")
+                    vlan_list += [str(i) for i in range(int(vlan_range[0]), int(vlan_range[-1]) + 1)]
+                else:
+                    vlan_list.append(str(vlan_part))
+
+            # break after first vlan line
+            break
+
+    return vlan_list
+
+def removeOldLines(config, interfaces, vlans):
+    config = config.splitlines()
+    new_config = []
+
+    # Remove lines that will be filled in by this plugin
+    delete_section = False
+    found_vlanrange_line = False
+    for line in config:
+        if line.startswith("!"):
+            # skip commands
+            continue
+
+        if line == "" and delete_section:
+            # Skip empty lines if deleting
+            continue
+
+        if not line.startswith(" "):
+            # This is a new section
+            delete_section = False
+
+        if line.startswith("interface breakout "):
+            # Existing breakout config, get port
+            match = re.search(r"port (\d+)", line)
+            if match:
+                port_num = match.group(1)
+                object_name = f"Ethernet1/{port_num}"
+                if object_name in interfaces:
+                    continue
+
+        if line.startswith("interface "):
+            # This is an interface config
+
+            # Find the object name
+            object_name = line.removeprefix("interface ")
+            if not object_name.startswith("Ethernet") and not object_name.startswith("port-channel"):
+                delete_section = False;
+                new_config.append(line)
+                continue
+
+            # Check if object is configured
+            if object_name.count("/") > 1:
+                # this is a breakout
```

Comment:
is this assuming that `Ethernet1/1` means the parent/whole interface but `Ethernet1/1/3` means a breakout interface? Is this mentioned in any nx-os documentation that this is a convention that's followed?

I vaguely remember cisco switches with interface names like `gi 1/0/1` and those weren't breakout ports. I think it used to donate slot or modules.

Reply:
Yeah we'll very likely never have devices with slots like that but it's a good point. I guess the best way then is to see if 1/29 already exists, and if 1/29/1 is found when 1/29 was already found then its a breakout